paasta_tools/setup_marathon_job.py

```diff
@@ -242,6 +243,11 @@ def log_bounce_action(line, level='debug'):
 
     kill_given_tasks(client=client, task_ids=[task.id for task in tasks_to_kill], scale=True)
 
+    for task in bounce_lib.flatten_tasks(old_app_at_risk_tasks):
+        if task in tasks_to_kill:
+            hostname = task.host
+            reserve_all_resources([hostname])
```

Comment:
I just wish it wasn't the bounce code's responsibility to do this.

How does this reservation even work if the bounce code hasn't killed these tasks yet?

Also I have a hunch that this needs some try/excepts. The bounce has to go on in the face of slow api endpoints and broken reservation stuff, in the context of reserving away resource for at-risk tasks. Heck, if it was easy I would say make this asynchronous and best-effort so the bounce can move on at normal speed!